ros/src/fastrack/include/fastrack/value/matlab_value_function.h

```diff
@@ -115,7 +119,17 @@ class MatlabValueFunction : public ValueFunction<TS, TC, TD, PS, PC, PD, B> {
   // Compute center of nearest grid cell to the given state.
   VectorXd NearestCenterPoint(const VectorXd& x) const;
 
-  // Recursive helper function for gradient multilinear interpolation.
+  // Recursive helper function for value or gradient multilinear interpolation.
+  // Takes in a state and index along which to interpolate, and a boolean
+  // determining whether this is for the gradient or, otherwise, for the value.
+  VectorXd RecursiveMultilinearInterpolator(const VectorXd& x, size_t idx,
```

Comment:
Ooooooh cool idea. Instead of passing in a `bool` flag, could template the function itself on the accessor function, like:
```
template <typename Accessor, typename OutputType>
OutputType RecursiveMultilinearInterpolator(const VectorXd& x, size_t idx) const;
```

Reply:
In any case, since the output type can be either `double` or `VectorXd` you need to template that.